api/types/network/network.go

```diff
@@ -21,8 +21,19 @@ const (
 
 // CreateRequest is the request message sent to the server for network create call.
 type CreateRequest struct {
-	CreateOptions
-	Name string // Name is the requested name of the network.
+	Name       string            // Name is the requested name of the network.
+	Driver     string            // Driver is the driver-name used to create the network (e.g. `bridge`, `overlay`)
+	Scope      string            // Scope describes the level at which the network exists (e.g. `swarm` for cluster-wide or `local` for machine level).
+	EnableIPv4 *bool             `json:",omitempty"` // EnableIPv4 represents whether to enable IPv4.
+	EnableIPv6 *bool             `json:",omitempty"` // EnableIPv6 represents whether to enable IPv6.
+	IPAM       *IPAM             // IPAM is the network's IP Address Management.
```

Comment:
As a follow-up, we should consider taking the opportunity to;

- Add `omitempty` labels to these fields
- Add Names for the json fields; we've been too relaxed on that, and it's probably good to eplicitly set the names (also with their expected casing).